toolchain/mfc/run/case_dicts.py

```
@@ -20,175 +60,221 @@
-        POST_PROCESS.append(f'{attribute}({cmp_id})')
+    for real_attr in ["mom_wrt", "vel_wrt", "flux_wrt", "omega_wrt"]:
+        add_param(f'{real_attr}({cmp_id})', POST_PROCESS, LOGICALS)
 
-for cmp_id in range(100):
-    POST_PROCESS.append(f'chem_wrt({cmp_id})')
+# NOTE: `chem_wrt` is missing
+# for cmp_id in range(100):
+#     POST_PROCESS.append(f'chem_wrt({cmp_id})')
 
 for fl_id in range(1,10+1):
-    for append in ["schlieren_alpha", "alpha_rho_wrt", "alpha_wrt", "kappa_wrt"]:
-        POST_PROCESS.append(f'{append}({fl_id})')
+    for append, ty in [("schlieren_alpha", REALS), ("alpha_rho_wrt", LOGICALS),
+                       ("alpha_wrt", LOGICALS), ("kappa_wrt", LOGICALS)]:
+        add_param(f'{append}({fl_id})', POST_PROCESS, ty)
 
-    for attribute in ["gamma", "pi_inf", "ss", "pv", "gamma_v", "M_v", "mu_v", "k_v", "G", "mul0",
+    for real_attr in ["gamma", "pi_inf", "ss", "pv", "gamma_v", "M_v", "mu_v", "k_v", "G", "mul0",
                       "cv", "qv", "qvp" ]:
-        POST_PROCESS.append(f"fluid_pp({fl_id})%{attribute}")
+        add_param(f"fluid_pp({fl_id})%{real_attr}", POST_PROCESS, REALS)
 
 ALL = list(set(PRE_PROCESS + SIMULATION + POST_PROCESS))
 
-CASE_OPTIMIZATION = [ "nb", "weno_order", "num_fluids"]
+CASE_OPTIMIZATION = [ "nb", "weno_order", "num_fluids" ]
+
+TYPED_SET = INTEGERS | REALS | LOGICALS | STRINGS
+assert TYPED_SET - set(ALL) == set(), "Found parameter without stage (COMMON, PREPROCESS, SIMULATION, POSTPROCESS)"
+assert set(ALL) - TYPED_SET  == set(), "Found untyped parameter!"
+
+_properties = {}
+for param in ALL:
+    if param in INTEGERS:
+        _properties[param] = {"type": "integer"}
+    elif param in REALS:
+        _properties[param] = {"type": "number"}
+    elif param in LOGICALS:
+        _properties[param] = {"enum": ["T", "F"]}
+    elif param in STRINGS:
+        _properties[param] = {"type": "string"}
+    else:
+        print(f'WARNING: Found parameter {param} without type!')
```

I think we should throw an error here instead of just a warning.

---

100% agree